### PR TITLE
Fix text update of SpinBox

### LINKED ISSUE #3047: manual input to SpinBox can't be accepted
Platform: Windows 10, 
Rust: 1.67-x86_64-pc-windows-msvc
Slint: 1.1.0

The slint file looks like 
```
export component MainWindow inherits Window{
    callback button-set-font(int);
    public function set-font-weight(w : float) {
        root.default-font-weight = w;
    }

    HorizontalLayout {
        spin-font-weight := SpinBox { value: 100; minimum: 1; maximum: 10000;}
        Button {
            text: "Set font weight"; 
            clicked => {
                button-set-font(spin-font-weight.value);
            }
        }
    }
}
```

and with rust code
```
    let main_weak_font = main_window.as_weak();
    main_window.on_button_set_font(move |curr|{
        let weight = curr;
        rfd::MessageDialog::new().set_description(&format!("set weight to {}", weight)).show();
        main_weak_font.unwrap().invoke_set_font_weight(weight as _);
    });
```

Evetytime I input some number to the spinbox and click the button, the message box would show other number.

### LINKED ISSUE #3053: SpinBox: value property sometimes fails to propagate to internal `TextInput`
When binding a `SpinBox`'s `value` to a property, then updating the property from outside of the `SpinBox`, the internal `TextInput`'s `value` is not updated in certain circumstances. I've experimented with the issue a bunch and I believe that the issue manifests when the value is updated from within the `SpinBox`, though I'm unsure if there are any other triggers.

Here are the steps I've found to reproduce the issue (you can follow along with this [SlintPad MRE][slintpad]):
1. Create a `SpinBox` with a binding to an integer property and some way for that property to be updated, either from another component or from native code.
2. Try updating the value externally (click one of the 3 buttons on the MRE) and notice that the value changes as expected.
3. Update the value using the `SpinBox`, either by editing the text input or by clicking one of the stepper buttons.
4. Try updating the value externally and notice that the value no longer updates.
5. Click a stepper button and notice that the `SpinBox` acts as though its value is what it had been externally updated to. This shows that the value within the `SpinBox` is indeed changing, but the `TextInput` is not receiving the new value.

I can reproduce this on the Qt backend (Qt 6, rust MSVC 1.70 stable, Windows 10 22H2 & Qt 5, rust GNU 1.70 stable, Debian 11 on WSL2) and the web backend (SlintPad).

Looking through the implementation for the Fluent-styled `SpinBox`, it seems that the issue stems from the way that the `TextInput`'s `text` prop is updated. Whenever the value is updated from within the `SpinBox`, `update-text()` is called to update the `TextInput` to match. However, when this is done, it seems as though some link between `root.value` and `i-text-input.text` is removed; perhaps `i-text-input.text` is set to a constant equivalent to whatever `root.value` was at the time?

I unfortunately don't know enough about Slint to suggest a good fix for this. My instinct would be to publicize the `update-value(int)` function but it would be rather weird for one singular standard component to require value updates to be performed through a function.

[slintpad]: https://slint.dev/releases/1.1.0/editor/?snippet=import+%7B+Button%2C+VerticalBox%2C+SpinBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++property%3Cint%3E+value+%3C%3D%3E+i-spin-box.value%3B%0A%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A++++++++i-spin-box+%3A%3D+SpinBox+%7B%0A++++++++++++minimum%3A+0%3B%0A++++++++++++maximum%3A+100%3B%0A++++++++%7D%0A++++++++Button+%7B%0A++++++++++++text%3A+%22Set+to+0%22%3B%0A++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++root.value+%3D+0%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++++++Button+%7B%0A++++++++++++text%3A+%22Set+to+50%22%3B%0A++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++root.value+%3D+50%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++++++Button+%7B%0A++++++++++++text%3A+%22Set+to+100%22%3B%0A++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++root.value+%3D+100%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A

### PR DESCRIPTION
**Pull request by FloVanGH**
Fixes #3053
Fixes #3047

### CODE REVIEW

**Comment by tronical** [comment]
Please include a regression test for this fix.

**Review by tronical — APPROVED** [review]

File: internal/compiler/widgets/common/spinbox-base.slint [hunk 1]
```diff
@@ -35,27 +33,39 @@ export component SpinBoxBase {
...
         vertical-alignment: center;
         horizontal-alignment: left;
         text: root.value;
         enabled: root.enabled;
         width: 100%;
         height: 100%;
 
+        accessible-role: AccessibleRole.text-input;
+        accessible-value: self.text;
+        accessible-label: "spinbox-textinput";
```
Comment (ogoffart) [inline internal/compiler/widgets/common/spinbox-base.slint:56]: I don't think that's a good use of accessible-label.

Now with this patch, I believe a screen reader would see two widgets with the same value, and one of them bing described as "spinbox-textinput" which will be hard for the user to understand.

Comment (FloVanGH) [reply internal/compiler/widgets/common/spinbox-base.slint:56]: I talked with Simon about that topic. He showed me with the accessibility inspector on macOS that it is accessibility wise spread in different parts.

Comment (tronical) [reply internal/compiler/widgets/common/spinbox-base.slint:56]: So this was just a workaround anyway, with the id based search we can remove this label again.

**Pull request merged**